lib/index.js

```diff
@@ -25,7 +25,7 @@ internals.schema = {
         registrations: Joi.array().items(Joi.object({
             plugin: [
                 Joi.string(),
-                Joi.object({ register: Joi.string() }).unknown()
+                Joi.object({ register: Joi.string(), options: Joi.object() }).unknown()
```

Comment:
While it's not explicitly checking `options`, it's not disallowing it either because `unknown()` is there. hapi allowing [anything](https://github.com/hapijs/hapi/blob/master/lib/schema.js#L351), I'd say this is already a proper schema for your use case. It could be improved to remove `unknown()` and allow `options: Joi.any()` but that's not what you're complaining about.